### ISSUE TITLE
[Bug] [1.8.01]: Files Renamed remove spaces in attachment (Solved), Kanban not working, TOC weird is contains link

### ISSUE THREAD

**Issue by roytoh** [KosmosisDire/obsidian-webpage-export#416]
### What happened?

Original file with Uppercase after export to web server turn lowercase.
All space turned to -

### Reproduction

When i export web server, it changed all files and attachment replace uppercase to lowercase and blank to -

### Version

1.8.01

### Operating System

Windows 11

### Obsidian Version

1.5.11

### Which browsers can you reproduce this in?

Microsoft Edge

### Relevant log output

```yaml
Log:
[INFO] 	Including stylesheet: css-settings-manager
[INFO] 	Including stylesheet: css-settings-manager
[INFO] 	Using "[[business]]" header as title because it was H1 at the top of the page
[INFO] 	Closing render window


Settings:
	settingsVersion --------- 1.8.01
	makeOfflineCompatible     false
	inlineAssets ------------ false
	includePluginCSS          4 plugins included
	includeSvelteCSS -------- true
	titleProperty             Grey Corporation
	customHeadContentPath ---
	faviconPath               F:\\\\- Database\\Grey Corp\\- icons\\Roy Toh.png
	documentWidth ----------- 40em
	sidebarWidth              20em
	minOutlineCollapse ------ 2
	startOutlineCollapsed     true
	allowFoldingHeadings ---- true
	allowFoldingLists         true
	allowResizingSidebars --- true
	logLevel                  warning
	minifyHTML -------------- true
	makeNamesWebStyle         true
	onlyExportModified ------ true
	deleteOldFiles            false
	addThemeToggle ---------- true
	addOutline                true
	addFileNav -------------- true
	addSearchBar              true
	addGraphView ------------ true
	addTitle                  true
	addRSSFeed -------------- false
	siteURL                   https://greycorp.ddns.net/
	authorName -------------- Roy Toh
	vaultTitle                Grey Corp
	exportPreset ------------ website
	openAfterExport           true
	graphAttractionForce ---- 1
	graphLinkLength           10
	graphRepulsionForce ----- 150
	graphCentralForce         3
	graphEdgePruning -------- 100
	graphMinNodeSize          3
	graphMaxNodeSize -------- 7
	showDefaultTreeIcons      true
	emojiStyle -------------- Native
	defaultFileIcon           lucide//file
	defaultFolderIcon ------- lucide//folder
	defaultMediaIcon          lucide//file-image
	exportPath -------------- F:\\Apache24\\htdocs
	filesToExport             2


Enabled Plugins:
	cMenu
	Update time on edit
	Find orphaned files and broken links
	Advanced Slides
	Buttons
	Commander
	Dataview
	Easy Typing
	Editor Syntax Highlight
	Emoji Toolbar
	Highlightr
	Iconize
	Image Toolkit
	Kanban
	Mind Map
	Omnisearch
	Outliner
	Projects
	Style Settings
	Templater
	ToggleList
	Html Server
	DB Folder
	MetaEdit
	Ribbon Divider
	Recent Files
	Table of Contents
	Diagrams.net
	Fast Image Cleaner by Roytoh
	Tag Wrangler
	Importer
	Webpage HTML Export
```

**Comment by aubreyz** [comment]
Which is exactly as it should be...

**Comment by KosmosisDire** [comment]
Yes, this is expected behavior. What is the reason you do not want this to happen? It only affects the file names, not the names displayed inside the document

**Comment by aubreyz** [comment]
> Because my obsidian around 30GB++, expert to web server crash. So i export without attachment then my attachment can't open because the linking in obsidian direct to files without space but my actual file with space.

Because in general you can't have spaces in file names in web files. This is nothing to do with this plugin. Likewise (in general but not an absolute rule) it is best to use only lower case file names.

**Comment by KosmosisDire** [comment]
Yes unfortunately it seems the real problem here is that the export crashes. However, if you would like to override this setting anyways you can navigate to the plugin's folder where the data.json file is stored, open the data.json as a text file, and set `makeNamesWebStyle` to false. This will do what you want, but isn't recommended normally.

**Comment by roytoh** [comment]
> Yes unfortunately it seems the real problem here is that the export crashes. However, if you would like to override this setting anyways you can navigate to the plugin's folder where the data.json file is stored, open the data.json as a text file, and set `makeNamesWebStyle` to false. This will do what you want, but isn't recommended normally.

**Problem solved.**
1. Currently i able to solve the issue of attachment has space while web server change link of attachment into without spaces
I use symbiolic link to link my current attachment file to the new exported web server in apache.
2. I found another plugin to enable my future attachment to remove space by editing their js which prevent future issue occurred
4. Current files i require manually rename to remove the spaces via obsidian app as i can't rename outside the obsidian software

**Issue Found**
1. I notice [kanban](obsidian://show-plugin?id=obsidian-kanban) can't display and scroll properly in exported web server
2. If heading is a link, the table of contents don't appear nicely as below photos
![image](https://github.com/KosmosisDire/obsidian-webpage-export/assets/85841486/b5e84b4f-5e63-4ea1-b8a8-52431dac03d1)

Issue to be verified.
1. Photos and videos i yet verified but photos seem working as i haven't rename all my files yet.

**Comment by KosmosisDire** [comment]
Your second two issues: 
1. Did you enable the kanban styles in the settings? If you want support for a plugin you have to enable it's styles.

2. This is a known issue that is fixed in the beta version

**Comment by roytoh** [comment]
![image](https://github.com/KosmosisDire/obsidian-webpage-export/assets/85841486/bfca3d6c-2966-41fb-b309-cd8fa3f01657)
1. The plugin i enable but it stucked and can't scroll in version 1.8.01 and 1.8.1-2b. I removed the kanban plugin as least can use as usual.
2. Solved

**1.8.1-2b** New Advantages
1. Click export button instead appeared compared previously need wait few minutes.
3. After appeared selection click export instantly appeared as well compared previously need wait few minutes.

**Comment by KosmosisDire** [comment]
I will fix the scrolling issue soon, I had noticed that as well

**Comment by roytoh** [comment]
> I will fix the scrolling issue soon, I had noticed that as well

Awesome thanks but i just tried the beta i can't export completely my files, i removed export attachment to reduce files exported. I try download back previous version and see if it works.

**Comment by KosmosisDire** [comment]
When you say you can't completely export your files what do you mean? I need more information to be able to know what is going wrong and how to fix it

**Comment by roytoh** [comment]
Hi,

I’ll try again and update later on the beta version.
I’ll include some screenshot of how the fails is about. I worry is from my
files because i export partial it works, full export it hangs.
Looking forward to your reply.

Thank you.

Best Wishes,

Roy.
+60105662093
t.me/lrtoh
www.roy-toh.com

**Comment by roytoh** [comment]
> When you say you can't completely export your files what do you mean? I need more information to be able to know what is going wrong and how to fix it

**Current Issues and Findings tested on beta 1.8.1-2b**
1. Kanban unable scroll
2. Export with delete function, the delete page stuck so long.
3. Click export button instantly appeared due to my obsidian 30GB++ while 1.8.01 load after few minutes.
4. When exporting my too many pages. it appeared below page.
![image](https://github.com/KosmosisDire/obsidian-webpage-export/assets/85841486/49f79eac-31a3-4c58-aa67-e4f16a84e1f2)

**Comment by KosmosisDire** [comment]
Can you check your memory usage in the task manager? Most likely obsidian is running out of memory or something

**Comment by KosmosisDire** [comment]
With such a huge vault there is only so much I can do to support it, but I do plan on experimenting with it exporting chunks at a time to reduce memory usage

**Comment by roytoh** [comment]
Currently i'm test using 1.8.01 to export, although it loads very long, but
it didn't disappeared.
At least currently it still works. but i do not export my attachment, only
export my .md files.
[image: image.png]

**Comment by roytoh** [comment]
I just tried 1.8.1-2b to export in chunk. It works but when my files goes big, it crash.

![image](https://github.com/KosmosisDire/obsidian-webpage-export/assets/85841486/c2ae464c-32c4-4812-b2fe-48c8e983bc96)

I think the solution available currently is either below option.
1. Split my vault into multiple vault
2. Get bigger RAM

**Comment by KosmosisDire** [comment]
The stable version of 1.9.0 is now out, if this is still present in the latest version please create a new issue
